### ISSUE TITLE
Build Fails on Mac OS 10.12.4: "function definition is not allowed here"

### ISSUE THREAD

**Issue by jackvandrunen** [zielmicha/reactor.nim#13]
```
Error: execution of an external compiler program 'clang -c  -w -pthread -I/Users/jack/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/include -I/Users/jack/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_DARWIN_USE_64_BIT_INODE=1  -I/usr/local/Cellar/nim/0.16.0/nim/lib -o /Users/jack/Documents/nim-cjdns/nimcache/cjdns.o /Users/jack/Documents/nim-cjdns/nimcache/cjdns.c' failed with exit code: 1

/Users/jack/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src/unix/async.c:242:60: error: function definition is not allowed here
void uv__async_stop(uv_loop_t* loop, struct uv__async* wa) {
                                                           ^
/Users/jack/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src/unix/async.c:258:32: error: function definition is not allowed here
static int uv__async_eventfd() {
                               ^
/Users/jack/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src/unix/async.c:298:2: error: expected '}'
}
 ^
/Users/jack/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src/unix/async.c:187:77: note: to match this '{'
int uv__async_start(uv_loop_t* loop, struct uv__async* wa, uv__async_cb cb) {
                                                                            ^
3 errors generated.
```

Code and complete traceback at https://gist.github.com/fallingduck/ce2a7e7f8cf39144e384d6eeb066a9e4

**Comment by girvo** [comment]
I'm also running up against this, which is a shame as I'm finding Reactor infinitely nicer to work with than the build in asyncdispatch/threadpool (and getting those two to combine is damned difficult! Doing so here, simple)

**Comment by girvo** [comment]
Forcing `cc: "gcc"` in the cfg gives the same error, so it's not `clang` that's doing it?

EDIT: I was completely wrong here, it was an error in the C code that `clang` complained about

**Comment by jackvandrunen** [comment]
@girvo Do you have `gcc` installed? On my machine, `gcc` is a symlink to `clang`. However, I did install `gcc` and some of Apple's libraries won't compile with it, so that was a dead end for me. At the moment I'm developing on a VM and on my server, so I'm working around it, but I'd still really like to see this resolved soon.

**Comment by girvo** [comment]
@fallingduck Oh hell, you're right. I do have it installed, but wrong binary. I forced it to use a real GCC:

```
        ... Error: execution of an external compiler program '/usr/local/bin/gcc-4.2 -c -w -pthread -I/Users/josh/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/include -I/Users/josh/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_DARWIN_USE_64_BIT_INODE=1  -I/Users/josh/Code/Nim/lib -o /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.o /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.c' failed with exit code: 1
        ... couldn't understand kern.osversion `16.5.0'
        ... /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.c: In function ‘atomicinc_4E9cxwuunx9aiL7vj9bdy8WvAsystem’:
        ... /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.c:7654: error: ‘__ATOMIC_RELAXED’ undeclared (first use in this function)
        ... /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.c:7654: error: (Each undeclared identifier is reported only once
        ... /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.c:7654: error: for each function it appears in.)
        ... /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.c: In function ‘atomicdec_4E9cxwuunx9aiL7vj9bdy8WvA_2system’:
        ... /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_system.c:7661: error: ‘__ATOMIC_RELAXED’ undeclared (first use in this function)
```

**Comment by girvo** [comment]
And using `gcc-7` (this is on macOS Sierra):

```
        ... CC: core
        ... Error: execution of an external compiler program '/usr/local/bin/gcc-7 -c -w -pthread -I/Users/josh/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/include -I/Users/josh/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_DARWIN_USE_64_BIT_INODE=1  -I/Users/josh/Code/Nim/lib -o /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_locks.o /Users/josh/PersonalProjects/Nim/binblog/src/nimcache/stdlib_locks.c' failed with exit code: 1
        ... In file included from /System/Library/Frameworks/Security.framework/Headers/AuthSession.h:32:0,
        ...                  from /System/Library/Frameworks/Security.framework/Headers/Security.h:43,
        ...                  from /System/Library/Frameworks/CoreServices.framework/Frameworks/OSServices.framework/Headers/CSIdentity.h:43,
        ...                  from /System/Library/Frameworks/CoreServices.framework/Frameworks/OSServices.framework/Headers/OSServices.h:27,
        ...                  from /System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/IconsCore.h:23,
        ...                  from /System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Headers/LaunchServices.h:22,
        ...                  from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:39,
        ...                  from /Users/josh/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src/unix/internal.h:52,
        ...                  from /Users/josh/.nimble/pkgs/reactor-0.4.1/reactor/uv/../../deps/libuv/src/threadpool.c:25:
        ... /System/Library/Frameworks/Security.framework/Headers/Authorization.h:192:7: error: variably modified 'bytes' at file scope
        ...   char bytes[kAuthorizationExternalFormLength];
        ...        ^~~~~
```

**Comment by girvo** [comment]
Okay, I've worked out what the issue is!

In `async.c`, line 226, there is a closing brace. The opening brace is not inside an `#ifdef`, but the closing brace is! Moving it after the `#endif` on 227 makes it compile perfectly :)

**Comment by jackvandrunen** [comment]
@girvo Be sure to submit the patch to @zielmicha 's fork of libuv!

**Comment by girvo** [comment]
@fallingduck I just went to do that, and it's already fixed his fork, but the commit that's referenced as a submodule is older than his master!

**Comment by zielmicha** [comment]
Fixed in v0.4.2.
